Site Editor: Update save panel's cancel button from icon to visible text

jonahtanjz opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
<!-- Please describe what you have changed or added -->
Resolves #29886

The current save panel for site editor uses an X icon to close the panel. This is inconsistent with the post editor's cancel button. This PR updates the save panel's cancel button of site editor from the X icon to the cancel button with text, similar to post editor's button.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
1. Open up site editor
2. Make any changes to the page
3. Click on the "Save" button to open up the save panel
4. The save and cancel buttons should be consistent with the design of the post editor panel's buttons

## Screenshots <!-- if applicable -->
| Before| After|
| ----------- | ----------- |
| ![image](https://user-images.githubusercontent.com/47470981/145699146-2c68f576-6216-468e-9f68-f75ed3edc3d2.png)      | ![image](https://user-images.githubusercontent.com/47470981/145699383-ed892dc3-9b51-4188-b906-097bf730a289.png)      |


## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

jonahtanjz committed:
Update site editor save panel's cancel button to visible text

kevin940726 reviewed (approved):
Thank you for making this PR! Code looks good and works as expected!

However, we could further tidy this up using `<Flex>`, `<FlexItem>` and `gap`. The JSX would look something like:

```jsx
<Flex className="entities-saved-states__panel-header" gap={ 2 }>
  <FlexItem
    isBlock
    as={ Button }
    { ...otherProps }
  >
    { __( 'Save' ) }
  </FlexItem>
  <FlexItem
    isBlock
    as={ Button }
    { ...otherProps }
  >
    { __( 'Cancel' ) }
  </FlexItem>
</Flex>
```

Then, we can delete all those custom CSS:

```diff
 .entities-saved-states__panel-header {
   /* Other styles... */
-  display: flex;
-  align-items: center;
-  align-content: space-between;
-
-  .components-button {
-    flex-grow: 1;
-    justify-content: center;
-    margin: auto;
-  }
-
-  .editor-entities-saved-states__cancel-button {
-    margin-left: 4px;
-  }
-
-  .editor-entities-saved-states__save-button {
-    margin-right: 4px;
-  }
 }
```

This could be a follow-up though 👍 .

kevin940726 commented on packages/editor/src/components/entities-saved-states/index.js:
```diff
@@ -197,10 +196,12 @@ export default function EntitiesSavedStates( { close } ) {
-					icon={ closeIcon }
+					className="editor-entities-saved-states__cancel-button"
 					onClick={ dismissPanel }
-					label={ __( 'Close panel' ) }
```
I wonder if we should keep this explicit label indicating that it's closing the panel. Could use some inputs from accessibility folks. WFYT @tellthemachines ?

jonahtanjz committed:
Update buttons to use Flex and FlexItem

jonahtanjz commented:
@kevin940726 Thanks for the review and suggestion! Since I'm already on this, I've changed it to use `<Flex>` and `<FlexItem>`, and also removed the custom CSS :)

kevin940726 commented on packages/editor/src/components/entities-saved-states/index.js:
```diff
@@ -221,13 +222,17 @@ export default function EntitiesSavedStates( { close } ) {
-					label={ __( 'Close panel' ) }
-				/>
-			</div>
+					className="editor-entities-saved-states__cancel-button"
```
I guess we don't need this anymore ;)

jonahtanjz replied:
👍

kevin940726 reviewed (commented):
Looks good! Thank you ♥️ !

jonahtanjz committed:
Remove unused class from cancel button

jeffpaul commented:
@jonahtanjz any chance you could confirm your WordPress.org account so that we can properly credit you in next week's WordPress 6.0 release?
